src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -919,7 +919,8 @@ private void WriteString(in DbRow row, Utf8JsonWriter writer)
         private static void Parse(
             ReadOnlySpan<byte> utf8JsonSpan,
             JsonReaderOptions readerOptions,
-            ref MetadataDb database)
+            ref MetadataDb database,
+            ref StackRowStack stack)
```

Comment:
Did this change of creating the `StackRowStack` up front and passing it along, and removing the `if (!IsInitialized)` from `Push` make a noticeable improvement in perf?

Reply:
Perhaps a tiny bit to avoid redundant `if` checks to see whether StackRowStack was initialized. But it does more closely align with the original code, simplified the logic in this method and avoided a `using` statement and indentation of almost the whole method.